pkg/reconciler/v1alpha1/taskrun/resources/input_resource_test.go

```diff
@@ -0,0 +1,313 @@
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: "git",
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/grafeas/kritis",
+				},
+				v1alpha1.Param{
+					Name:  "ServiceAccount",
+					Value: "kritis-service-account",
+				},
+			},
+		},
+	}
+	pipelineResourceInformer.Informer().GetIndexer().Add(res)
+	pipelineResourceInformer.Informer().GetIndexer().Add(resWithServiceAccount)
+}
+
+func TestAddResourceToBuild(t *testing.T) {
+	boolTrue := true
+
+	task := &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "build-from-repo",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Sources: []v1alpha1.Source{
+					v1alpha1.Source{
+						Name: "workspace",
+						Type: "git",
+					},
+				},
+			},
+		},
+	}
+	taskRun := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "build-from-repo-run",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name: "simpleTask",
+			},
+			Inputs: v1alpha1.TaskRunInputs{
+				Resources: []v1alpha1.PipelineResourceVersion{
+					v1alpha1.PipelineResourceVersion{
+						ResourceRef: v1alpha1.PipelineResourceRef{
+							Name: "workspace",
+						},
+						Version: "master",
+					},
+				},
+			},
+		},
+	}
+	build := &buildv1alpha1.Build{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Build",
+			APIVersion: "build.knative.dev/v1alpha1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "build-from-repo",
+			Namespace: "marshmallow",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "pipeline.knative.dev/v1alpha1",
+					Kind:               "TaskRun",
+					Name:               "build-from-repo-run",
+					Controller:         &boolTrue,
+					BlockOwnerDeletion: &boolTrue,
+				},
+			},
+		},
+		Spec: buildv1alpha1.BuildSpec{},
+	}
+
+	for _, c := range []struct {
+		desc    string
+		task    *v1alpha1.Task
+		taskRun *v1alpha1.TaskRun
+		build   *buildv1alpha1.Build
+		wantErr bool
+		want    *buildv1alpha1.Build
+	}{{
+		desc:    "simple",
+		task:    task,
+		taskRun: taskRun,
+		build:   build,
+		wantErr: false,
+		want: &buildv1alpha1.Build{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Build",
+				APIVersion: "build.knative.dev/v1alpha1"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "build-from-repo",
+				Namespace: "marshmallow",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "pipeline.knative.dev/v1alpha1",
+						Kind:               "TaskRun",
+						Name:               "build-from-repo-run",
+						Controller:         &boolTrue,
+						BlockOwnerDeletion: &boolTrue,
+					},
+				},
+			},
+			Spec: buildv1alpha1.BuildSpec{
+				Source: &buildv1alpha1.SourceSpec{
+					Git: &buildv1alpha1.GitSourceSpec{
+						Url:      "https://github.com/grafeas/kritis",
+						Revision: "master",
```

Comment:
can you add a task run  with `Revision` other than  master specified for completeness.